airflow/api/common/delete_dag.py

```diff
@@ -54,6 +54,15 @@ def delete_dag(dag_id: str, keep_records_in_log: bool = True, session=None) -> i
     if dag is None:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
+    # deleting a DAG should also all of its subdags
+    dags_to_delete = [
+        d[0]
+        for d in session.query(DagModel.dag_id)
+        .filter(and_(DagModel.dag_id.like(dag_id + ".%"), DagModel.is_subdag))
+        .all()
+    ]
+    dags_to_delete.append(dag_id)
```

Comment:
```suggestion
    dags_to_delete_query = session.query(DagModel.dag_id).filter(
        or_(
            and_(DagModel.dag_id.like(f"{dag_id}.%"), DagModel.is_subdag),
            DagModel.dag_id == dag_id
        )
    )
    dags_to_delete = [dag_id for dag_id, in dags_to_delete_query]
```

Would this be better?

Reply:
Yeah, I think so - would `dags_to_delete` be a list of string here, or a list of one-element tuples? I always get mixed up with the return types from SQLAlchemy queries 🤔

Reply:
Yup, this totally works, I've updated my changes.

Reply:
`list(dags_to_delete_query)` would have been a list of one-tuples, but the extra comma after `dag_id` unpacks the tuple and turn `dags_to_delete` into a list of strings.